src/Compilers/CSharp/Portable/Symbols/ConstructedNamedTypeSymbol.cs

```diff
@@ -89,7 +92,7 @@ internal override ImmutableArray<TypeWithAnnotations> TypeArgumentsWithAnnotatio
         {
             get
             {
-                return _typeArgumentsWithAnnotations;
+                return _typeArgumentsWithAnnotations.ToImmutable();
```

Comment:
I would be concerned about this causing more array allocations, not less, in the single type argument case. Since it is quite common to get the type arguments through this accessor.

Reply:
Agree but this property is defined in the base class and is referenced in +140 places, refactor everything or create a new property to access the internal object? if other any suggestion? Just let me know the right way, if any, and I will implement it

Reply:
Neither?  I'm not sure this approach makes sense in the first place.  Having 30 MB of allocs here doesn't seem problematic.

Reply:
Ok, let me know if you want me to help on this issue in the future